### PR TITLE
feat(rhineng-10859): Add decorator to ensure org_id integrity

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

This PR is being created to address [RHINENG-10859](https://issues.redhat.com/browse/RHINENG-10859).
* Create a decorator that process the response of API calls and validates that it only contains data for the current org
* Apply decorator to APIs

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: lib/middleware.py [hunk 1]
```diff
@@ -175,3 +178,40 @@ def rbac_group_id_check(rbac_filter: dict, requested_ids: set) -> None:
...
+                        logger.error(message)
+                        abort(HTTPStatus.FORBIDDEN)
+                if "results" in data:
+                    results_array = data.get("results", [])
+                    for system in results_array:
+                        org_id = system.get("org_id")
+                        if org_id and (org_id != current_identity.org_id):
+                            logger.error(message)
+                            abort(HTTPStatus.FORBIDDEN)
+            return make_response(jsonify(data), status_code)
```
Comment (vkrizan) [inline lib/middleware.py:213]: Does this decorator need to redo the response? I'd personally keep it just checking the correctness and abort if necessary and leave it return what the decorated function returned without any change.

Comment (chambridge) [reply lib/middleware.py]: make sense. Updated this to just return the original `result`.

File: lib/middleware.py [hunk 2]
```diff
@@ -175,3 +176,38 @@ def rbac_group_id_check(rbac_filter: dict, requested_ids: set) -> None:
...
+    def decorator(func):
+        @wraps(func)
+        def decorator_method(*args, **kwargs):
+            current_identity = get_current_identity()
+            result = func(*args, **kwargs)
+
+            if isinstance(result, Response):
+                data = result.get_json()
+            elif isinstance(result, tuple):
+                data, _ = result
```
Comment (vkrizan) [inline lib/middleware.py:191]:
```suggestion
                data = result[0]
```

**Review by vkrizan — APPROVED** [review]

**Comment by roliveri** [comment]
Would it make sense to be able to enable/disable this via a flag?

**Review by FabriciaDinizRH — APPROVED** [review]
LGTM

**Review by kruai — APPROVED** [review]
Thanks for this. LGTM

**Review by thearifismail — APPROVED** [review]
LGTM

**Pull request merged**